optimade/validator/utils.py

```diff
@@ -365,16 +361,15 @@ def wrapper(
                     success_type = "optional" if optional else None
                     validator.results.add_success(f"{request} - {msg}", success_type)
             else:
-                request = request.replace("\n", "")
+                request = request.replace("\n", "")  # type: ignore[union-attr]
                 message = msg.split("\n")
                 if validator.verbosity > 1:
                     # ValidationErrors from pydantic already include very detailed errors
                     # that get duplicated in the traceback
                     if not isinstance(result, ValidationError):
                         message += traceback.split("\n")
 
-                message = "\n".join(message)
-
+                failure_type = None
```

Comment:
I am not so happy with this line. The value of failure_type will be overwritten later on, so this assignment looks rather strange.
Is it such a problem to define the type on what is now line 377?

Reply:
Not a problem, but not of any importance, I don't think. I'd rather have the default value outside of the clause rather than a type-hint inside the clause. Not everything has to be explicitly typed (and we shouldn't let mypy dictate how we code beyond any real errors, especially as it changes between versions).